docs/clients/aws/ec2/ec2.md

```diff
@@ -0,0 +1,262 @@
+
+Finally the [`relabeling_configs`][relabel] section has three purposes:
+
+1. Selecting the labels discovered you want to attach to your targets. In our case here, we're keeping `instance_id` as instance, the tag `Name` as name and the `zone` of the instance. Make sure to check out the Prometheus [documentation][ec2_sd_config] for the full list of available labels.
+
+2. Choosing where promtail should find log files to tail, in our example we want to include all log files that exist in `/var/log` using the glob `/var/log/**.log`. If you need to use multiple glob, you can simply add another job in your `scrape_configs`.
+
+3. Ensuring discovered targets are only for the machine Promtail currently runs on. This is achieve by adding the label `__host__` using the incoming metadata `__meta_ec2_private_dns_name`. If it doesn't match the current `HOSTNAME` environnement variable, the target will be dropped.
+
+Alright we should be ready to fire up promtail, we're going to run it using the flag `--dry-run`. This is perfect to ensure everything is correctly, specially when you're still playing around with the configuration. Don't worry when using this mode, Promtail won't send any logs and won't remember any file positions.
+
+```bash
+ ./promtail-linux-amd64 -config.file=./ec2-promtail.yaml --dry-run
+```
+
+If everything is going well Promtail should print out log lines with their labels discovered instead of sending them to Loki, like shown below:
+
+```bash
+2020-07-08T14:51:38	{filename="/var/log/cloud-init.log", instance="i-041b0be05c2d5cfad", name="promtail-demo", zone="us-east-2c"}	Jul 07 21:37:24 cloud-init[3035]: util.py[DEBUG]: loaded blob returned None, returning default.
+```
+
+Don't hesitate to edit the your config file and start Promtail again to try your config out.
+
+If you want to see existing targets and available labels you can reach promtail server using the public dns assigned to your instance:
+
+```bash
+open http://ec2-13-59-62-37.us-east-2.compute.amazonaws.com:3100/
+```
+
+For example the page below is the service discovery page. It shows you all discovered targets, with their respective available labels and the reason it was dropped if it was the case.
+
+![discovery page page][discovery page]
+
+This page is really useful to understand what labels are available to forward with the `relabeling` configuration but also why Promtail is not scraping your target.
+
+## Configuring Promtail as a service
+
+Now that we have correctly configured Promtail. We usually want to make sure it runs as a [systemd service][systemd], so it can automatically restart on failure or when the instance restart.
+
+Let's create a new service using `vim /etc/systemd/system/promtail.service` and copy the service definition below:
+
+```systemd
+[Unit]
+Description=Promtail
+
+[Service]
+User=root
+WorkingDirectory=/opt/promtail/
+ExecStartPost=/bin/sleep 30
```

Comment:
why this?

Reply:
the AWS metadata endpoint might not ready when the node start.

Reply:
I think you need ExecStartPre then?